crates/turbo-tasks-macros/src/value_trait_macro.rs

```diff
@@ -265,6 +278,12 @@ pub fn value_trait(args: TokenStream, input: TokenStream) -> TokenStream {
                 }
             }
 
+            impl turbo_tasks::FromSubTrait<#ref_ident> for #supertrait_refs {
```

Comment:
In my `Vc<T>` work, I called this trait `Upcast` (for both the `Vc<dyn Sub>` -> `Vc<dyn Super>` and the `Vc<Val> -> Vc<dyn Trait>` cases).